test/function_tests/Uart/test_Uart_read.c

```diff
@@ -0,0 +1,31 @@
+#include <stdbool.h>
+#include "Uart.h"
+
+static int timeoutLimit = 0;
+
+static Uart_ErrorCode errCode = Uart_ErrorCode_OK;
+
+static uint8_t text[] = "Read text (sync) - data will be read until 'q' or 20 "
+                        "chars are received:\r\n";
+
+bool
+test_Uart_read(Uart* uart)
+{
+    uint8_t buf = '\0';
+    uart->txFifo = NULL;
```

Comment:
It's not needed in this scenario but if this test will be used in different order then not disabling the fifo will return false failure. I think it isn't good practice to assume hard requirement of order in which tests should be performed if it's not necessary.

Reply:
You are corrent that the tests should be independent from each other. Usually, in a test framework, this is done using setup and teardown methods/functions.
I would expect that setup would do Uart_init and Uart_startup, while teardown would do Uart_shutdown.
As we are not using a test framework, and hence we are initializing the environment for each test, it would be prefferable to call Uart_init and Uart_startup, instead of manipulating the internal data.
And in fact, your current code does rely on test order - because test_write must be called after test_init, right?

Reply:
done